### PR TITLE
Added generation of default currency.json

### LINKED ISSUE #186: Create currency.json if it is not found on startup
Add the functionality to create the `currency.json` file if it is not found on startup.

### PR DESCRIPTION
**Pull request by krusagiz**
Added generation of default currency.json when the program startups if there is no valid one already in place.

Resolves #186

### CODE REVIEW

File: src/main/java/thrift/MainApp.java [hunk 1]
```diff
@@ -160,8 +161,22 @@ protected void initCurrencyMappings(Storage storage) {
             logger.warning("Currency Mappings file at " + currencyMappingsFilePath
                     + " is not in the correct format. "
                     + "Using default currency mappings");
+
+            Map<String, Double> defaultCurrencyMappings = CurrencyUtil.getCurrencyMap();
+            try {
+                storage.saveCurrencyMappings(defaultCurrencyMappings);
+            } catch (IOException ioe) {
+                //do nothing
```
Comment (limerencee) [inline src/main/java/thrift/MainApp.java]: Should throw an exception and possibly a `logger.error()`

Comment (limerencee) [reply src/main/java/thrift/MainApp.java]: Also, is there a more elegant way than having a nested try-catch block? 🤔

File: src/main/java/thrift/MainApp.java [hunk 2]
```diff
@@ -160,8 +161,22 @@ protected void initCurrencyMappings(Storage storage) {
...
+            }
         } catch (IOException | NoSuchElementException e) {
             logger.warning("Problem while reading from the file. Using default currency mappings");
+
+            Map<String, Double> defaultCurrencyMappings = CurrencyUtil.getCurrencyMap();
+            try {
+                storage.saveCurrencyMappings(defaultCurrencyMappings);
+            } catch (IOException ioe) {
+                //do nothing
+            }
```
Comment (limerencee) [inline src/main/java/thrift/MainApp.java]: Repeated code, should do some form of abstraction 😅

**Review by limerencee — CHANGES_REQUESTED** [review]
Should do minor abstraction then it is good to merge.

File: src/main/java/thrift/MainApp.java (hunk 1 above)
Comment (krusagiz) [reply src/main/java/thrift/MainApp.java]: The logic has been changed to save the file regardless of whether a valid file exists or not.

File: src/main/java/thrift/MainApp.java (hunk 2 above)
Comment (krusagiz) [reply src/main/java/thrift/MainApp.java]: The logic has been changed to save the file regardless of whether a valid file exists or not. The code no longer requires repetition.

**Review by limerencee — APPROVED** [review]
LGTM!

**Pull request merged**